client-app/src/Components/NewItemForm.tsx

```diff
@@ -74,62 +141,64 @@ const NewItemForm: React.FC = () => {
                 [name]: base64Images,
             }));
         } else {
-            setFormData(prevState => ({
-                ...prevState,
-                [name]: value,
-            }));
+            // For dropdowns, we store the donor and program ID
+            if (name === 'itemType') {
+                const selectedType = itemTypeOptions.find(
+                    option => option.value === value,
+                );
+                setFormData(prevState => ({
+                    ...prevState,
+                    itemType: value,
+                }));
+            }
+            if (name === 'donorEmail') {
+                const selectedDonor = donorEmailOptions.find(
+                    option => option.value === value,
+                );
+                setFormData(prevState => ({
+                    ...prevState,
+                    donorEmail: value,
+                    donorId: selectedDonor?.id || null, // Store donor ID
+                }));
+            } else if (name === 'program') {
+                const selectedProgram = programOptions.find(
+                    option => option.value === value,
+                );
+                setFormData(prevState => ({
+                    ...prevState,
+                    program: value,
+                    programId: selectedProgram?.id || null, // Store program ID
+                }));
+            } else {
+                setFormData(prevState => ({
+                    ...prevState,
+                    [name]: value,
+                }));
+            }
         }
-        setErrors(prevState => ({ ...prevState, [name]: '' })); // Reset errors on change
+        setErrors(prevState => ({ ...prevState, [name]: '' }));
         setErrorMessage(null);
         setSuccessMessage(null);
     };
 
-    // Generalized validation function to reduce code repetition
+    // Validation for individual fields
     const validateField = (name: string, value: any) => {
         const requiredFields = [
             'itemType',
             'currentStatus',
             'donorEmail',
             'program',
-            'imageUpload',
             'dateDonated',
         ];
-
         if (requiredFields.includes(name)) {
-            if (name === 'itemType') {
-                if (!value || value.length === 0) {
-                    return 'Please select an item type';
-                }
-            } else if (name === 'currentStatus') {
-                if (!value || value.length === 0) {
-                    return 'Please enter a status';
-                }
-            } else if (name === 'donorEmail') {
-                if (!value || value.length === 0) {
-                    return "Please select the donor's email";
-                }
-            } else if (name === 'program') {
-                if (!value || value.length === 0) {
-                    return 'Please select a program';
-                }
-            } else if (name === 'imageUpload') {
```

Comment:
Please show different message for `imageUpload`, when count exceeds 5, show the relevant error. Please do not touch the old code for this PR